magma/testing/utils.py

```diff
@@ -125,3 +126,21 @@ def non_standard_operation(self):
         v0 = self._val << 2
         v1 = bits(self._val[0], len(self.T)) << 1
         return SimpleMagmaProtocol(v0 | v1 | bits(self._val[0], len(self.T)))
+
+
+
+def with_config(key, value):
+
+    def deco(fn):
+
+        @functools.wraps(fn)
+        def wrapper(*args, **kwargs):
+            prev_value = getattr(config, key)
+            setattr(config, key, value)
+            ret = fn(*args, **kwargs)
+            setattr(config, key, prev_value)
+            return ret
+
+        return wrapper
+
+    return deco
```

Comment:
One thing, when using the Python fixture approach with explicit setup/teardown stages, this should handle the case when the test throws an uncaught exception.  Otherwise, pytest won't know to run the teardown code even when the test fails (not normally a problem, but in theory this could cause downstream tests to also fail if the config state isn't reverted after one of these tests fail)

Reply:
:+1: using that now